atlas-2024-odfr/mkjson.py

```diff
@@ -0,0 +1,210 @@
+                my_files = []
+                for afile in my_files_dict:
+                    my_files += [ {'filename':afile,
+                                   'checuksum':my_files_dict[afile]['checksum'],
+                                   'size':my_files_dict[afile]['size'],
+                                   'events':my_files_dict[afile]['events'],
+                                   'type':my_files_dict[afile]['type'],
+                                   'uri_root':my_files_dict[afile]['uri'] } ]
+                # Final check that we have at least one file
+                if len(my_files)==0:
+                    print(f'No files identified for {dataset_line.strip()}')
+                    continue
+            else:
+                # This is a list of metadata for the MC samples
+                filename = 'MC_'+dataset_line.split()[2].strip().replace('\\','')+'_filelist.json'
+                # Get the list of files
+                my_did = dataset_line.split()[0].strip()
+                my_full_did_names = [ x for x in mc_json['file_locations'] if '.'+my_did+'.' in x ]
+                if len(my_full_did_names)>1:
+                    print(f'Warning: Found multiple matching DIDs from {my_did} : {my_full_did_names}')
+                elif len(my_full_did_names)==0:
+                    print(f'Found no matching DIDs from {my_did} : {my_full_did_names}')
+                    continue
+                my_full_did_name = my_full_did_names[0]
+                # Grab the dictionary of file metadata from the json file
+                my_files_dict = mc_json['file_locations'][my_full_did_name]
+                # Convert the dictionary into the format we want for the open data portal
+                my_files = []
+                for afile in my_files_dict:
+                    my_files += [ {'filename':afile,
+                                   'checuksum':my_files_dict[afile]['checksum'],
+                                   'size':my_files_dict[afile]['size'],
+                                   'events':my_files_dict[afile]['events'],
+                                   'type':my_files_dict[afile]['type'],
+                                   'uri_root':my_files_dict[afile]['uri'] } ]
+                # Final check that we have at least one file
+                if len(my_files)==0:
+                    print(f'No files identified for {my_full_did_name} from {my_did}')
+                    continue
+
+            # Set the metadata in the `super` (open data portal record) json
+            my_json['files'] += [ { 'filename':filename } ]
+            # Now open that file and write the file names there
+            with open(output_directory+'/'+filename,'w') as dataset_filelist_file:
+                json.dump( my_files , dataset_filelist_file )
+
+    # Write myself a json file
+    with open(output_directory+'/'+adataset.replace('.txt','.json'),'w') as outfile:
+        json.dump( my_json , outfile )
```

Comment:
When writing final record JSON files in the output directory such as `test/pp_2015_data_p6026_tids.json` or `test/mc_exotics_nominal.json`, it would be great if these files are directly open-data-repository-loadable as fixtures.  This means that the files should contain list of records, even if the list contains only one record.  In other words, something like this:

```json
[
  {
    "abstract": {
      "description": "MC simulation QCD jet nominal samples from the ATLAS experiment"
    },
    "accelerator": "CERN-LHC",
    "collaboration": {
      "name": "ATLAS collaboration"
    },
    "collections": [
      "ATLAS-MC-Simulation-Datasets"
    ],
    "collision_information": {
      "energy": "13TeV",
      "type": "pp"
    },
...
    "usage": {
      "description": "<p> The data and MC simulation provided by the ATLAS experiment in DAOD_PHYSLITE format is released under a CC0-BY license. This format can be used directly like a ROOT ntuple for simple studies or processed into secondary ntuples with systematic uncertainties included using the ATLAS AnalysisBase software. <p>Extensive instructions for interacting with the data, as well as documentation of the dataset naming conventions and their contents, are provided on the ATLAS Open Data website linked below.",
      "links": [
        {
          "description": "ATLAS Open Data Website",
          "url": "http://opendata.atlas.cern"
        },
        {
          "description": "Citation of ATLAS Open Data",
          "url": "https://opendata.atlas.cern/docs/documentation/ethical_legal/citation_policy"
        },
        {
          "description": "ATLAS Analysis Software Tutorial",
          "url": "https://atlassoftwaredocs.web.cern.ch/ASWTutorial/TutorialWeek/"
        }
      ]
    }
  }
]
```

In other words, you could write a list containing my_json, which will do the job.

Note also that the final record JSON files are usually produced with alphabetically-ordered fields and follow two-space indentation.  This is usually achieved via a `json.dumps()` call like:

```python
        json.dumps(
            records,
            indent=2,
            sort_keys=True,
            ensure_ascii=False,
            separators=(",", ": "),
        )
```

See also helper script https://github.com/cernopendata/opendata.cern.ch/blob/master/scripts/clean_json_file.py that can do this on the produced file.

Reply:
Done!